File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/cpp_to_py/code_output/SignInSystem.py
class SignInSystem:
    def __init__(self):
        self.users = {}

    def add_user(self, username):
        if username in self.users:
            return False
        else:
            self.users[username] = False
            return True

    def sign_in(self, username):
        if username not in self.users:
            return False
        else:
            self.users[username] = True
            return True

    def check_sign_in(self, username):
        if username not in self.users:
            return False
        else:
            return self.users[username]

    def all_signed_in(self):
        for value in self.users.values():
            if not value:
                return False
        return True

    def all_not_signed_in(self):
        return [username for username, value in self.users.items() if not value]

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_cpp/SignInSystem.cpp
#include <unordered_map>
#include <vector>
#include <string>

class SignInSystem {
private:
    std::unordered_map<std::string, bool> users;

public:
    SignInSystem() {}

    bool add_user(const std::string& username) {
        if (users.find(username) != users.end()) {
            return false;
        } else {
            users[username] = false;
            return true;
        }
    }

    bool sign_in(const std::string& username) {
        auto it = users.find(username);
        if (it == users.end()) {
            return false;
        } else {
            it->second = true;
            return true;
        }
    }

    bool check_sign_in(const std::string& username) const {
        auto it = users.find(username);
        if (it == users.end()) {
            return false;
        } else {
            return it->second;
        }
    }

    bool all_signed_in() const {
        for (const auto& pair : users) {
            if (!pair.second) {
                return false;
            }
        }
        return true;
    }

    std::vector<std::string> allNotSignedIn() const {
        std::vector<std::string> notSignedInUsers;
        for (const auto& pair : users) {
            if (!pair.second) {
                notSignedInUsers.push_back(pair.first);
            }
        }
        return notSignedInUsers;
    }
};